### PR TITLE
added cloudformation code

### PR DESCRIPTION
**Pull request by nour-saad**
I added the cloudformation code to create a VPC on AWS, check it 🔥

### CODE REVIEW

File: AWS-VPC_deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,113 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: VPC with 2 public subnets and 2 private subnets
+
+Resources:
+  VPC:
+    Type: AWS::EC2::VPC
+    Properties:
+      CidrBlock: 10.0.0.0/16
```
Comment (enkhalifapro) [inline AWS-VPC_deploy.yaml]: Use parameterized values instead of static values

File: AWS-VPC_deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+          Value: AWS-VPC
+
+  InternetGateway:
+    Type: AWS::EC2::InternetGateway
+    Properties:
+      Tags:
+        - Key: Name
+          Value: MyIGW
+
+  VPCGatewayAttachment:
```
Comment (enkhalifapro) [inline AWS-VPC_deploy.yaml]: Rename it in a way that shows the gateway type. Example, `InternetGatewayAttachment`

File: AWS-VPC_deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref VPC
+      CidrBlock: 10.0.4.0/24
+      AvailabilityZone: !Select [1, !GetAZs ""]
+      Tags:
+        - Key: Name
+          Value: PrivateSubnet2
+
+  RouteTable:
```
Comment (enkhalifapro) [inline AWS-VPC_deploy.yaml]: Rename it to show its purpose. Example, `PublicRouteTable`

File: AWS-VPC_deploy.yaml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref VPC
+      Tags:
+        - Key: Name
+          Value: PublicRouteTable
+
+  PublicRoute:
```
Comment (enkhalifapro) [inline AWS-VPC_deploy.yaml]: It won't be the only route to be attached to the table, hence give it more specific name. Example, `DefaultPublicRoute`

**Review by enkhalifapro — CHANGES_REQUESTED** [review]
Important:
You missed to add:
- NateGateways
- ElasticIPs
- PrivateRouteTables 
- Parameters Section
- Outputs Section

File: AWS-VPC_deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,214 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: VPC with 2 public subnets and 2 private subnets
+
+#Usage: aws cloudformation create-stack --stack-name MyVPCStack --template-body file://AWS-VPC_deploy.yaml --parameters ParameterKey=VPC_CidrBlock,ParameterValue="x.x.x.x/x" ParameterKey=PubSub1CidrBlock,ParameterValue="x.x.x.x/x" ParameterKey=PubSub2CidrBlock,ParameterValue="x.x.x.x/x" ParameterKey=PrivSub1CidrBlock,ParameterValue="x.x.x.x/x" ParameterKey=PrivSub2CidrBlock,ParameterValue="x.x.x.x/x" --capabilities CAPABILITY_NAMED_IAM
```
Comment (enkhalifapro) [inline AWS-VPC_deploy.yaml]: clean any dummy code up

File: AWS-VPC_deploy.yaml (hunk 2 above)
Comment (enkhalifapro) [reply AWS-VPC_deploy.yaml]: It's still the same comment! refer to line 53

File: AWS-VPC_deploy.yaml [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+      VpcId: !Ref VPC
+      Tags:
+        - Key: Name
+          Value: PublicRouteTable
+
+# Creating internet route and associating it to the InternetGateway
+  DefaultPublicInternetRoute:
+    Type: AWS::EC2::Route
+    Properties:
+      RouteTableId: !Ref PublicRouteTable
```
Comment (enkhalifapro) [inline AWS-VPC_deploy.yaml:124]: You Ref to another component while you are not sure it is generated or no. Use DependOn attribute to solve this problem

Comment (enkhalifapro) [reply AWS-VPC_deploy.yaml:124]: This comment is general for all components that depend on other ones

File: AWS-VPC_deploy.yaml [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+
+  PublicSubnet2RouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      SubnetId: !Ref PublicSubnet2
+      RouteTableId: !Ref PublicRouteTable
+
+# Creating NAT Gateway and EIP for each public subnet
+  EIPForNat1:
```
Comment (enkhalifapro) [inline AWS-VPC_deploy.yaml:146]: You need to add Name tag
```
      Tags:
        - Key: Name
          Value: NatGateway1IP
```

**Review by enkhalifapro — CHANGES_REQUESTED** [review]
You are still missing:
- Private routes

File: AWS-VPC_deploy.yaml [hunk 8]
```diff
@@ -0,0 +1,264 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: VPC with 2 public subnets and 2 private subnets
+
+#######################
```
Comment (enkhalifapro) [inline AWS-VPC_deploy.yaml]: Remove the dummy code

File: AWS-VPC_deploy.yaml (hunk 2 above)
Comment (enkhalifapro) [reply AWS-VPC_deploy.yaml]: It's still

File: AWS-VPC_deploy.yaml [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+        Value: PublicSubnet2RouteTableAssociation
+
+# Creating NAT Gateway and EIP for each public subnet
+  EIPForNat1:
+    Type: AWS::EC2::EIP
+    Properties:
+      Domain: vpc
+    Tags:
+      - Key: Name
+        Value: EIPForNat1
```
Comment (enkhalifapro) [inline AWS-VPC_deploy.yaml]: Name tag should be descriptive

File: AWS-VPC_deploy.yaml (hunk 8 above)
Comment (nour-saad) [reply AWS-VPC_deploy.yaml]: removed dummy code

File: AWS-VPC_deploy.yaml (hunk 2 above)
Comment (nour-saad) [reply AWS-VPC_deploy.yaml]: renamed

File: AWS-VPC_deploy.yaml (hunk 6 above)
Comment (nour-saad) [reply AWS-VPC_deploy.yaml:124]: added DependsOns

File: AWS-VPC_deploy.yaml (hunk 7 above)
Comment (nour-saad) [reply AWS-VPC_deploy.yaml:146]: added name tags

File: AWS-VPC_deploy.yaml (hunk 9 above)
Comment (nour-saad) [reply AWS-VPC_deploy.yaml]: changed name tag to EIPForNatGatewayX

**Pull request closed without merging**